src/zipcrypto.rs

```diff
@@ -0,0 +1,158 @@
+use std::num::Wrapping;
+
+struct ZipCryptoKeys {
+    key_0: Wrapping<u32>,
+    key_1: Wrapping<u32>,
+    key_2: Wrapping<u32>,
+}
+
+impl ZipCryptoKeys {
+    //Used this paper to implement ZipCrypto algo
+    //https://courses.cs.ut.ee/MTAT.07.022/2015_fall/uploads/Main/dmitri-report-f15-16.pdf
+
+    fn new() -> ZipCryptoKeys {
+        ZipCryptoKeys {
+            key_0: Wrapping(0x12345678),
+            key_1: Wrapping(0x23456789),
+            key_2: Wrapping(0x34567890),
+        }
+    }
+
+    fn update(&mut self, input: u8) {
+        self.key_0 = ZipCryptoKeys::crc32(self.key_0, input);
+        self.key_1 =
+            (self.key_1 + (self.key_0 & Wrapping(0xff))) * Wrapping(0x08088405) + Wrapping(1);
+        self.key_2 = ZipCryptoKeys::crc32(self.key_2, (self.key_1 >> 24).0 as u8);
+    }
+
+    fn stream_byte(&mut self) -> u8 {
+        let temp: Wrapping<u16> = Wrapping(self.key_2.0 as u16) | Wrapping(3);
+        ((temp * (temp ^ Wrapping(1))) >> 8).0 as u8
+    }
+
+    fn decrypt_byte(&mut self, cipher_byte: u8) -> u8 {
+        let plain_byte: u8 = self.stream_byte() ^ cipher_byte;
+        self.update(plain_byte);
+        plain_byte
+    }
+
+    #[allow(dead_code)]
+    fn encrypt_byte(&mut self, plain_byte: u8) -> u8 {
+        let cipher_byte: u8 = self.stream_byte() ^ plain_byte;
+        self.update(plain_byte);
+        cipher_byte
+    }
+
+    fn crc32(crc: Wrapping<u32>, input: u8) -> Wrapping<u32> {
+        return (crc >> 8) ^ Wrapping(CRCTABLE[((crc & Wrapping(0xff)).0 as u8 ^ input) as usize]);
+    }
+}
+
+pub struct ZipCryptoReader<R> {
+    file: R,
+    keys: ZipCryptoKeys,
+}
+
+impl<R: std::io::Read> ZipCryptoReader<R> {
+    pub fn new(file: R, password: &[u8]) -> ZipCryptoReader<R> {
+        //Note: The password is &[u8] and not &str because the documentation
```

Comment:
nit: comments should have a space between the `//` and the first character

Reply:
Fixed.